lms/djangoapps/courseware/access.py

```diff
@@ -618,13 +618,18 @@ def can_load():
         students to see modules.  If not, views should check the course, so we
         don't have to hit the enrollments table on every module load.
         """
+        sneakpeek_access = ACCESS_DENIED
         if user.is_authenticated():
             if not UserProfile.has_registered(user):
                 if _can_load_descriptor_nonregistered(descriptor):
-                    return ACCESS_GRANTED
+                    sneakpeek_access = ACCESS_GRANTED
         response = (
             _visible_to_nonstaff_users(descriptor)
-            and _has_group_access(descriptor, user, course_key)
+            and
+            (
+                _has_group_access(descriptor, user, course_key)
+                or sneakpeek_access
```

Comment:
> I'm still seeing Section 3 in the screenshot.

By section 3 I mean the 3rd section which is called: **Example Week 2: Get Interactive**

DIrect Access users **do** have an account, but they haven't registered so it fails `_has_group_access`